pkg/proxy/hybrid/proxy.go

```diff
@@ -14,17 +14,31 @@ import (
 	apihelper "k8s.io/kubernetes/pkg/api/helper"
 	kcorelisters "k8s.io/kubernetes/pkg/client/listers/core/internalversion"
 	"k8s.io/kubernetes/pkg/proxy"
-	"k8s.io/kubernetes/pkg/proxy/config"
 
 	unidlingapi "github.com/openshift/origin/pkg/unidling/api"
 	"k8s.io/kubernetes/pkg/proxy/userspace"
 )
 
+// EndpointsConfigHandler is an abstract interface of objects which receive update notifications for the set of endpoints.
+type EndpointsConfigHandler interface {
+	// OnEndpointsUpdate gets called when endpoints configuration is changed for a given
+	// service on any of the configuration sources. An example is when a new
+	// service comes up, or when containers come up or down for an existing service.
+	OnEndpointsUpdate(endpoints []*api.Endpoints)
+}
+
+// ServiceConfigHandler is an abstract interface of objects which receive update notifications for the set of services.
+type ServiceConfigHandler interface {
+	// OnServiceUpdate gets called when a configuration has been changed by one of the sources.
+	// This is the union of all the configuration sources.
+	OnServiceUpdate(services []*api.Service)
+}
+
 type HybridProxier struct {
 	unidlingProxy        *userspace.Proxier
-	unidlingLoadBalancer config.EndpointsConfigHandler
-	mainEndpointsHandler config.EndpointsConfigHandler
-	mainServicesHandler  config.ServiceConfigHandler
+	unidlingLoadBalancer EndpointsConfigHandler
+	mainEndpointsHandler EndpointsConfigHandler
+	mainServicesHandler  ServiceConfigHandler
```

Comment:
It looks like the upstream types were renamed to `EndpointsHandler` and `ServiceHandler`. This is going to need a bunch of work though because the interfaces have grown and HybridProxier needs to implement the new functions. I guess you already filed #15101 for this.